### ISSUE TITLE
crypto-dev-linux not compatible anymore to linux mainline kernel 

### ISSUE THREAD

**Issue by redbrain17** [cryptodev-linux/cryptodev-linux#23]
crypto_alloc_ablkcipher() was removed from linux kernel code inbetween Kernel 4.7-4.8 

This seems for me the commit which removed this function:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/patch/crypto/ablkcipher.c?id=3a01d0ee2b991c8c267620e63a4ab47cd8c30cc4

This is the Build Error I got if I try to use it with a recent Linux Kernel: 

```
|   CC [M]  /home/graute/cryptodev-module/ioctl.o
|   CC [M]  /home/graute/cryptodev-module/main.o
|   CC [M]  /home/graute/cryptodev-module/cryptlib.o
| /home/graute/cryptodev-module/cryptlib.c: In function 'cryptodev_cipher_init':
| /home/graute/cryptodev-module/cryptlib.c:138:18: error: implicit declaration of function 'crypto_alloc_ablkcipher' [-Werror=implicit-function-declaration]
|    out->async.s = crypto_alloc_ablkcipher(alg_name, 0, 0);
|                   ^
| /home/graute/cryptodev-module/cryptlib.c:138:16: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
|    out->async.s = crypto_alloc_ablkcipher(alg_name, 0, 0);
|                 ^
| cc1: some warnings being treated as errors
| make[3]: *** [/home/graute/cryptodev-module/cryptlib.o] Error 1
| make[2]: *** [_module_/home/graute/cryptodev-module] Error 2
| make[2]: Leaving directory `/home/graute/5411_IBIS/yocto/build-imx6ulevk-nand/tmp/work-shared/imx6ulevk/kernel-build-artifacts'
| make[1]: *** [sub-make] Error 2
| make[1]: Leaving directory `/home/graute/5411_IBIS/yocto/build-imx6ulevk-nand/tmp/work-shared/imx6ulevk/kernel-source'
| make: *** [build] Error 2
| ERROR: oe_runmake failed

```

**Comment by cristian-stoica** [comment]
The current version of cryptodev should build and work with both old and recent kernels, including 4.7 and 4.8

**Comment by cristian-stoica** [comment]
fixed post v1.9 (a705360197260d28535746ae98c461ba2cfb7a9e)
